### DISCUSSION TITLE
輪読レビュー時に追記する箇所について

### DISCUSSION THREAD

**Discussion by RyoyaC** [code-application/reading-notebook#2] [General]
## レビューの基本的な流れ
[この形式](https://github.com/orgs/lean-ja/discussions/8)のように、PR上で

> 1. 担当を決める
> 2. 担当者は、担当箇所についてsorryを埋めて、プルリクエストを投げる
> 3. 参加者全員でレビューして、議論がなくなったらマージする
> 4. 以上を繰り返す

を行う。

### 輪読レビュー時に追記する箇所
このとき、3について、議論の実施について疑問点が出たので相談です。

> 参加者全員でレビューして、議論がなくなったらマージする

片方がノートにコメント書いて、もう片方が別の意見（もしくは同意見）のとき

- そこのコメントに追記する
- 各章のノートのDiscussionに移動して論点をまとめる
- PR上で会話して終わらせる

などいくつかやり方があるかなと思っています。

原則会話したこと自体のエピソード記憶で思い出すこともあるから、そうしたコメントなども残しておきたい、というのがコメントを残す意義なので、コメントに追記する、もしくはDiscussionに移動する、がいいかと思ってます。

個人的には、どちらかがコメントを出したら、もう一方が異なる意見を述べた場合、それをDiscussionで整理する流れが良いと思います。
ただ、ノート内のDiscussionにまとめるとき、ただの会話を議事メモのように残すと論点が見にくくなると思っていて、Discussionに移動するときは論点整理して載せた方がいいとも思っています。

それを踏まえると、そうした会話形式のコメントをどこまで残そうかというところですが、

- コメント→会話形式でそのまま残し、返信（別意見、同意見など）は>などで続ける。
- ディスカッション→コメントで出てきた論点を整理し、後で見返しやすくする

のような感じにするのはいかがでしょうか？

**Comment by Kattsuuya** [comment]
ノート、コメント、Discussionはそれぞれ何を指してますか…？
（念の為意味を揃えたく🙏）

**Reply by RyoyaC** [reply]
以下の想定です
- ノート：本の各章のファイル（マークダウン）
- コメント：ノート内に記載する[!NOTE]などで記載する感想など
- Discussion：（GitHubのDiscussionsではなく）ノート内に記載するDiscussion（今はまだない）

**Reply by Kattsuuya** [reply]
@RyoyaC 
ありがとうございます！
理解できました。

私としては、輪読会の時間内に議論した内容については千葉さんに提案していただいた方針でまとめることに賛成です！
一方で、輪読会の時間外に読んで後日PRを投げた部分についてはコメントに追記するに留める形でお願いしたいと考えてます。

この書籍に関してはまずは知らない概念をインプットすることに注力したい、という気持ちが背景にあります😎

**Reply by RyoyaC** [reply]
@Kattsuuya 
> この書籍に関してはまずは知らない概念をインプットすることに注力したい、という気持ちが背景にあります

一応意図としては、コメントに追記＋議論した内容は整理する、という感じで時系列情報＋論点整理というイメージだったんですが、今は本の内容自体のインプット優先で、コメントなどの付加情報の整理は劣後したい、っていう意味であってますか？

**Reply by Kattsuuya** [reply]
@RyoyaC 
非同期であとからPRを出すと論点を整理するタイミングはない認識なので、整理するなら輪読会で行いたいです！
もしかしたら僕のイメージがズレてるかもなので、次集まったとき話させてください🙇

**Reply by RyoyaC** [reply]
@Kattsuuya 
> 論点を整理するタイミングはない認識

ここは普通に仕事でスライド作ったりするときみたいに非同期で論点整理してまとめるのでよいかなと思ってました。
